rust-runtime/aws-smithy-http-server/src/proto/aws_json/rejection.rs

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use strum_macros::Display;
```

Comment:
Yeah let's derive `thiserror::Error`. We should also sprinkle `tracing::debug!(%err)` all throughout the codebase whenever returning a rejection type; `debug` though, since these are not errors.

I'd like to refactor some things to remove some of these variants first.

Reply:
Yeah, `tracing::debug!` would be better.

https://github.com/awslabs/smithy-rs/issues/2521